src/Microsoft.ML.AutoML/ColumnInference/ColumnTypeInference.cs

```diff
@@ -357,12 +361,18 @@ private static InferenceResult InferTextFileColumnTypesCore(MLContext context, I
             // validate & retrieve label column
             var labelColumn = GetAndValidateLabelColumn(args, cols);
 
-            // if label column has all Boolean values, set its type as Boolean
-            if (labelColumn.HasAllBooleanValues())
+            // if label column has all Boolean values and allowBinary is true, set its type as Boolean
+            if (labelColumn.HasAllBooleanValues() && allowBinary)
             {
                 labelColumn.SuggestedType = BooleanDataViewType.Instance;
             }
 
+            // if labelColumn has Boolean values and allowBianry is false, set its type to Text because it's raw type must be string (or it will be parse as single).
+            if(labelColumn.SuggestedType == BooleanDataViewType.Instance && !allowBinary)
+            {
+                labelColumn.SuggestedType = TextDataViewType.Instance;
+            }
```

Comment:
I don't think the right answer is passing down a single use "allowBinary". This is meant to be more generic.

Is there a cleaner way? Perhaps could add a `ToString` transform to the pipeline to modify? Or perhaps modify the type in the returned `InferenceResult `?